src/sdf/include/sdf.h

```diff
@@ -0,0 +1,474 @@
+}
+
+__host__ __device__ glm::ivec3 TetTri0(int i) {
+  constexpr glm::ivec3 tetTri0[16] = {{-1, -1, -1},  //
+                                      {0, 3, 4},     //
+                                      {0, 1, 5},     //
+                                      {1, 5, 3},     //
+                                      {1, 4, 2},     //
+                                      {1, 0, 3},     //
+                                      {2, 5, 0},     //
+                                      {5, 3, 2},     //
+                                      {2, 3, 5},     //
+                                      {0, 5, 2},     //
+                                      {3, 0, 1},     //
+                                      {2, 4, 1},     //
+                                      {3, 5, 1},     //
+                                      {5, 1, 0},     //
+                                      {4, 3, 0},     //
+                                      {-1, -1, -1}};
+  return tetTri0[i];
+}
+
+__host__ __device__ glm::ivec3 TetTri1(int i) {
+  constexpr glm::ivec3 tetTri1[16] = {{-1, -1, -1},  //
+                                      {-1, -1, -1},  //
+                                      {-1, -1, -1},  //
+                                      {3, 4, 1},     //
+                                      {-1, -1, -1},  //
+                                      {3, 2, 1},     //
+                                      {0, 4, 2},     //
+                                      {-1, -1, -1},  //
+                                      {-1, -1, -1},  //
+                                      {2, 4, 0},     //
+                                      {1, 2, 3},     //
+                                      {-1, -1, -1},  //
+                                      {1, 4, 3},     //
+                                      {-1, -1, -1},  //
+                                      {-1, -1, -1},  //
+                                      {-1, -1, -1}};
+  return tetTri1[i];
+}
+
+__host__ __device__ glm::ivec4 Neighbors(int i) {
+  constexpr glm::ivec4 neighbors[14] = {{0, 0, 0, 1},     //
+                                        {1, 0, 0, 0},     //
+                                        {0, 1, 0, 0},     //
+                                        {0, 0, 1, 0},     //
+                                        {-1, 0, 0, 1},    //
+                                        {0, -1, 0, 1},    //
+                                        {0, 0, -1, 1},    //
+                                        {-1, -1, -1, 1},  //
+                                        {-1, 0, 0, 0},    //
+                                        {0, -1, 0, 0},    //
+                                        {0, 0, -1, 0},    //
+                                        {0, -1, -1, 1},   //
+                                        {-1, 0, -1, 1},   //
+                                        {-1, -1, 0, 1}};
+  return neighbors[i];
+}
+
+__host__ __device__ Uint64 SpreadBits3(Uint64 v) {
+  v = v & 0x1fffff;
```

Comment:
These magic numbers look really magical, it would be nice if you can explain how/where did you get them.

Reply:
I have similar ones in the collider; Morton codes are pretty standard and various versions of the magic numbers can be found in algorithm reference books.